### PR TITLE
Add Advanced Door Functionality.

### PR DESCRIPTION
**Pull request by bradknez**
Doors now:
- Can tell if they're open or closed.
- Can take the player from room to room.
- Take the player to the proper room.
- Store information about which direction they go and which room to enter.

### CODE REVIEW

**Comment by bjhobson12** [comment]
Well done Brad! I'll let the second reviewer merge, but I'm working off of your branch now!

**Comment by bjhobson12** [comment]
@bradknez  Did you remember to merge master into your own branch prior to this PR? I don't see magic numbers.

**Comment by bradknez** [comment]
@bjhobson12 I merged with master just before the PR, there’s some in the classes I’ve added but that is because for the most part they’re not finished quite yet, hope that’s okay!

File: Sprintfinity3902/Dungeon/Dungeon.cs [hunk 1]
```diff
@@ -134,12 +135,56 @@ public void SetCurrentRoom(int id)
         {
             NextId = id;
         }
+        public void ChangeRoom(IDoor door)
```
Comment (kylegessel) [inline Sprintfinity3902/Dungeon/Dungeon.cs:138]: Could this be a separate class in the future?

File: Sprintfinity3902/Entities/Items/ArrowItem.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        Vector2 currentRect;
+
+        public ArrowItem(Vector2 position)
+        {
+            Sprite = ItemSpriteFactory.Instance.CreateArrowVerticalItem();
+            Sprite2 = ItemSpriteFactory.Instance.CreateArrowHorizontalItem();
+            Position = position;
+            itemUse = false;
+            itemUseCount = 0;
+            // TODO: Change when items fully implemented.
```
Comment (kylegessel) [inline Sprintfinity3902/Entities/Items/ArrowItem.cs:29]: What is the change needed here?

File: Sprintfinity3902/PauseMenu.cs [hunk 3]
```diff
@@ -163,6 +164,14 @@ public void ChangePosition()
                 }
             }
 
+            foreach (IEntity door in Game.dungeon.GetCurrentRoom().doors)
+            {
+                if (count != 176 * Global.Var.SCALE && Pause)
+                    door.Y = door.Y + 2 * Global.Var.SCALE;
+                else if (count != 176 * Global.Var.SCALE && Pause == false)
+                    door.Y = door.Y - 2 * Global.Var.SCALE;
```
Comment (kylegessel) [inline Sprintfinity3902/PauseMenu.cs:172]: Magic numbers need to be fixed here.

**Review by kylegessel — APPROVED** [review]
Looks solid, going to do some brief testing! Makes it feel like a real game.

In testing, I found the up/down doors harder to enter than the left and right doors. Not necessarily a bug, just something I've noticed.

File: Sprintfinity3902/PauseMenu.cs (hunk 3 above)
Comment (kylegessel) [reply Sprintfinity3902/PauseMenu.cs:172]: Looks like the entire PauseMenu.cs still needs a magic numbers rework so I'm fine with merging this.

**Pull request merged**